### PR TITLE
Consume Tooling API dependency directly from the build

### PR DESCRIPTION
**Pull request by donat**

### CODE REVIEW

File: org.gradle.toolingapi/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,78 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: Gradle Tooling API
+Bundle-Vendor: Gradle Inc.
+Bundle-SymbolicName: org.gradle.toolingapi
+Bundle-ClassPath: .
+Bundle-RequiredExecutionEnvironment: JavaSE-1.7
+Require-Bundle: org.slf4j.api;bundle-version="1.7.2"
+Export-Package: org.gradle;version="4.3.0",
```
Comment (oehme) [inline org.gradle.toolingapi/META-INF/MANIFEST.MF]: ❌ Almost none of the packages below are part of the tooling API. We should only export the ones that are actually part of the official API.

File: org.gradle.toolingapi/.project [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
```
Comment (oehme) [inline org.gradle.toolingapi/.project]: ❌ I'm a bit confused by the whole setup. Why is the TAPI jar checked in? Shouldn't that be the job of the build to set up the project?

Comment (donat) [reply org.gradle.toolingapi/.project]: The jar is only needed to make the Eclipse workspace compile successfully. Without it, we'd need to execute Gradle tasks when setting up the dev environment.

File: org.gradle.toolingapi/META-INF/MANIFEST.MF (hunk 1 above)
Comment (donat) [reply org.gradle.toolingapi/META-INF/MANIFEST.MF]: I didn't change anything in that regard. I remember we had problems before when not all packages were exported. IMO we should tackle this separately.

Comment (oehme) [reply org.gradle.toolingapi/META-INF/MANIFEST.MF]: Can you open an issue for that for 3.0? We definitely need to fix this, as these packages should not be exposed to anyone.

Comment (donat) [reply org.gradle.toolingapi/META-INF/MANIFEST.MF]: https://github.com/eclipse/buildship/issues/671

**Pull request merged**